src/rules/__tests__/no-deprecated-flash.test.js

```diff
@@ -0,0 +1,201 @@
+      sx={{fontSize: 0, borderRadius: 0, py: 2, px: 3}}
+      className="custom-class"
+    >
+      Banner content
+    </Banner>
+  )
+}`,
+      errors: [{messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}],
+    },
+
+    // Mixed imports - Flash with other components
+    {
+      code: `import {Button, Flash, Text} from '@primer/react'
+
+function Component() {
+  return (
+    <div>
+      <Button>Click me</Button>
+      <Flash variant="error">Error message</Flash>
+      <Text>Some text</Text>
+    </div>
+  )
+}`,
+      output: `import {Button, Text} from '@primer/react'
+import {Banner} from '@primer/react/experimental'
+
+function Component() {
+  return (
+    <div>
+      <Button>Click me</Button>
+      <Banner variant="error">Error message</Banner>
+      <Text>Some text</Text>
+    </div>
+  )
+}`,
+      errors: [{messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}],
+    },
+
+    // Flash only import
+    {
+      code: `import {Flash} from '@primer/react'
+
+function Component() {
+  return <Flash>Just Flash</Flash>
+}`,
+      output: `import {Banner} from '@primer/react/experimental'
+
+function Component() {
+  return <Banner>Just Flash</Banner>
+}`,
+      errors: [{messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}],
+    },
+
+    // Self-closing Flash
+    {
+      code: `import {Flash} from '@primer/react'
+
+function Component() {
+  return <Flash variant="info" />
+}`,
+      output: `import {Banner} from '@primer/react/experimental'
+
+function Component() {
+  return <Banner variant="info" />
+}`,
+      errors: [{messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}],
+    },
+
+    // Multiple Flash components
+    {
+      code: `import {Flash} from '@primer/react'
+
+function Component() {
+  return (
+    <div>
+      <Flash variant="warning">Warning</Flash>
+      <Flash variant="error">Error</Flash>
+    </div>
+  )
+}`,
+      output: `import {Banner} from '@primer/react/experimental'
+
+function Component() {
+  return (
+    <div>
+      <Banner variant="warning">Warning</Banner>
+      <Banner variant="error">Error</Banner>
+    </div>
+  )
+}`,
+      errors: [{messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}],
+    },
+
+    // Flash with existing Banner import (should not duplicate)
+    {
+      code: `import {Flash} from '@primer/react'
+import {Banner} from '@primer/react/experimental'
+
+function Component() {
+  return (
+    <div>
+      <Flash variant="warning">Flash message</Flash>
+      <Banner variant="info">Banner message</Banner>
+    </div>
+  )
+}`,
+      output: `import {Banner} from '@primer/react/experimental'
+
+function Component() {
+  return (
+    <div>
+      <Banner variant="warning">Flash message</Banner>
+      <Banner variant="info">Banner message</Banner>
+    </div>
+  )
+}`,
+      errors: [{messageId: 'flashDeprecated'}, {messageId: 'flashDeprecated'}],
+    },
+  ],
+})
```

Comment:
Also consider a case where other stuff is already being imported from `primer/react/experimental` like `TooltipV2`